index.js

```diff
@@ -127,18 +127,21 @@ module.exports = {
         var decryptedMessage = nacl.box.open(ciphertext, nonce, ephemPublicKey, recieverEncryptionPrivateKey);
 
         // return decrypted msg data
-        var output = nacl.util.encodeUTF8(decryptedMessage);
+        try {
+          var output = nacl.util.encodeUTF8(decryptedMessage);
```

Comment:
This is a question we should dive into: What should the format of the message be? I don't see why it should always be UTF8. I think a lot of times we would rather use a binary format, for instance when encrypting keys etc. Since we are using `nacl` anyway we should probably stick to the `UInt8Array` format for the message and then leave it up to the implementer to encode it into the proper format.

I don't know though if we want to have an interface that accept hex-strings etc to conform to `web3` norms, so that's something to keep in mind.

Reply:
Ok after discussions with @danfinlay I’m now ok with using UTF8 as a format for the cleartext message. The reason is that the RPC endpoint will always use stringified JSON as a format, so we can assume that the messages passed in are of this format as well.